aptos-move/move-examples/veiled_coin/sources/veiled_coin.move

```diff
@@ -0,0 +1,376 @@
+
+        deposit(recipient, vc)
+    }
+
+    /// Returns the commitment to the balance of `owner` for the provided `CoinType`.
+    public entry fun private_balance<CoinType>(owner: address): CompressedRistretto acquires VeiledCoinStore {
+        assert!(
+            has_veiled_coin_store<CoinType>(owner),
+            error::not_found(EVEILED_COIN_STORE_NOT_PUBLISHED),
+        );
+
+        borrow_global<VeiledCoinStore<CoinType>>(owner).private_balance
+    }
+
+    //
+    // Private functions.
+    //
+
+    /// Returns a signer for the resource account.
+    fun get_resource_account_signer(): signer acquires VeiledCoinMinter {
+        account::create_signer_with_capability(&borrow_global<VeiledCoinMinter>(@veiled_coin).signer_cap)
+    }
+
+    /// Used internally to drop veiled coins that were split or joined.
+    fun drop_veiled_coin<CoinType>(c: VeiledCoin<CoinType>) {
+        let VeiledCoin<CoinType> { private_value: _ } = c;
+    }
+
+    //
+    // Public functions.
+    //
+
+    /// Returns `true` if `account_addr` is registered to receive veiled coins of `CoinType`.
+    public fun has_veiled_coin_store<CoinType>(account_addr: address): bool {
+        exists<VeiledCoinStore<CoinType>>(account_addr)
+    }
+
+    /// Returns the commitment to the value of `coin`.
+    public fun private_value<CoinType>(coin: &VeiledCoin<CoinType>): &Commitment {
+        &coin.private_value
+    }
+
+    /// Returns true if the balance at address `owner` equals `value`, which should be useful for auditability. Requires
+    /// the Pedersen commitment randomness as an auxiliary input.
+    public fun verify_opened_balance<CoinType>(owner: address, value: u64, randomness: &Scalar): bool acquires VeiledCoinStore {
+        // compute the expected committed balance
+        let value = new_scalar_from_u64(value);
+        let expected_comm = pedersen::new_commitment_for_bulletproof(&value, randomness);
+
+        // get the actual committed balance
+        let actual_comm = pedersen::new_commitment_from_compressed(&private_balance<CoinType>(owner));
+
+        pedersen::commitment_equals(&actual_comm, &expected_comm)
+    }
+
+    /// Initializes a veiled coin store for the specified account.
+    public fun register<CoinType>(account: &signer) {
+        let account_addr = signer::address_of(account);
+        assert!(
+            !has_veiled_coin_store<CoinType>(account_addr),
+            error::already_exists(EVEILED_COIN_STORE_ALREADY_PUBLISHED),
+        );
+
+        let coin_store = VeiledCoinStore<CoinType> {
+            private_balance: ristretto255::point_identity_compressed(),
+            deposit_events: account::new_event_handle<DepositEvent>(account),
+            withdraw_events: account::new_event_handle<WithdrawEvent>(account),
+        };
+        move_to(account, coin_store);
+    }
+
+    /// Mints a veiled coin from a normal coin, shelving the normal coin into the resource account's coin store.
```

Comment:
Is moving an account's coin into the same account's coin store equivalent to locking it up?

Reply:
Eh, not sure what you meant here, but I think by now you have answered this for yourself?